src/main/java/org/opensearch/knn/plugin/stats/suppliers/ModelIndexStatusSupplier.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.knn.plugin.stats.suppliers;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.ResourceNotFoundException;
+import org.opensearch.cluster.health.ClusterHealthStatus;
+import org.opensearch.knn.index.memory.NativeMemoryCacheManager;
+import org.opensearch.knn.indices.ModelDao;
+
+import java.util.Locale;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+public class ModelIndexStatusSupplier<T> implements Supplier<T> {
+    public static Logger logger = LogManager.getLogger(ModelIndexStatusSupplier.class);
+    private Function<ModelDao, T> getter;
+
+    /**
+     * Constructor
+     *
+     * @param getter ModelDAO Method to supply a value
+     */
+    public ModelIndexStatusSupplier(Function<ModelDao, T> getter) {
+        this.getter = getter;
+    }
+
+    @Override
+    public T get() {
+        try{
+            return getter.apply(ModelDao.OpenSearchKNNModelDao.getInstance());
+        } catch (Exception e) { //catch to prevent exception to be raised.
```

Comment:
Cant we catch the IndexNotFoundException as opposed to general Exception?

Reply:
Ack.